youtube_dl/extractor/ciscolive.py

```diff
@@ -137,35 +137,25 @@ def _check_bc_id_exists(rf_item):
         return int_or_none(try_get(rf_item, lambda x: x['videos'][0]['url'])) is not None
 
     def _entries(self, query, url):
-        query['size'] = 50
-        query['from'] = 0
-        for page_num in itertools.count(1):
-            results = self._call_api(
-                'search', None, query, url,
-                'Downloading search JSON page %d' % page_num)
-            if int(results['totalSearchItems']) == 0:
-                raise ExtractorError(
-                    'Search api returned no items (if matches are expected rfApiProfileId may be invalid)',
-                    expected=True)
-            sl = try_get(results, lambda x: x['sectionList'][0], dict)
-            if sl:
-                results = sl
-            items = results.get('items')
-            if not items or not isinstance(items, list):
-                break
-            for item in items:
-                if not isinstance(item, dict):
-                    continue
-                if not self._check_bc_id_exists(item):
-                    continue
-                yield self._parse_rf_item(item)
-            size = int_or_none(results.get('size'))
-            if size is not None:
-                query['size'] = size
-            total = int_or_none(results.get('total'))
-            if total is not None and query['from'] + query['size'] > total:
-                break
-            query['from'] += query['size']
+        results = self._call_api(
+            'search', None, query, url,
+            'Downloading search JSON')
+        if int(results['totalSearchItems']) == 0:
+            raise ExtractorError(
+                'Search api returned no items (if matches are expected rfApiProfileId may be invalid)',
+                expected=True)
+        sl = try_get(results, lambda x: x['sectionList'], list)
+        if sl is not None:
+            for s in sl:
+                items = s.get('items')
+                if not items or not isinstance(items, list):
+                    break
+                for item in items:
+                    if not isinstance(item, dict):
+                        continue
+                    if not self._check_bc_id_exists(item):
+                        continue
+                    yield self._parse_rf_item(item)
```

Comment:
You could import `traverse_obj` and `T` from utils to make this safer/simpler:
```suggestion
        if traverse_obj(results, ('totalSearchItems', T(int_or_none)), default=0) == 0:
            raise ExtractorError(
                'Search api returned no items (if matches are expected rfApiProfileId may be invalid)',
                expected=True)
        for item in traverse_obj(results, (
                # expository comment that needn't be in the updated code:
                # go here, expand list, go here, expand matching list items 
                'sectionList', Ellipsis, 'items', lambda _, v: self._check_bc_id_exists(v))):
            yield self._parse_rf_item(item)
```